### ISSUE TITLE
Analyze and test the architectures of neural networks used in existing research.

### ISSUE THREAD

**Issue by WojtAcht** [agh-a2s/nn-surrogate-benchmark#12]
Examine how the size of a neural network affects its ability to approximate the original problem.

A comprehensive repository: https://github.com/GMC-DRL/psc4MetaBBO

[Locality-aware Surrogates for Gradient-based Black-box Optimization](https://arxiv.org/pdf/2501.19161)
<img width="1149" alt="Image" src="https://github.com/user-attachments/assets/cdd86bd9-62b5-4bb2-9c15-732314f8cda3" />

Neural Networks as Black-Box Benchmark Functions Optimized for Exploratory Landscape Feature
```{python}
model = nn.Sequential(
    nn.Linear(dim, 512),
    nn.Tanh(),
    nn.Linear(512, 1),
    nn.Sigmoid()
)
optimizer = AdamW(model.parameters(), lr=0.001)
```

[Conservative Objective Models for Effective Offline Model-Based Optimization](https://arxiv.org/pdf/2107.06882)
<img width="679" alt="Image" src="https://github.com/user-attachments/assets/463c2322-07ca-4a41-8b70-a69e790387e8" />
